common/network-common/src/main/java/org/apache/spark/network/client/TransportClientFactory.java

```diff
@@ -245,12 +245,13 @@ TransportClient createClient(InetSocketAddress address)
     logger.debug("Creating new connection to {}", address);
 
     Bootstrap bootstrap = new Bootstrap();
+    int connCreateTimeout = conf.connectionCreationTimeoutMs();
```

Comment:
I would have simply thrown an exception when `connCreateTimeout <= 0` - and we can avoid the rest of the PR.
Will that not work ?

Reply:
w/o this PR, here we simply throw an exception, it doesn't work

Reply:
Yes, it will throw the exception and user will fix invalid config first time they run

Reply:
Not the same; checkvalue fails app in driver ahead and tells users what is invalid. He we fail during executor allocating. Users may not encounter exceptions, or exceptions and config are suppressed

Reply:
This failure is in `createClient` (when introduced) - and will fail all invocations for that module - which will in turn fail the task/stage/job. This is similar in behavior to a configuration referenced in executor which fails in `checkValue`.
The failure reason will inform the user about the invalid configuration.

Do you have a scenario where an exception thrown in this codepath does not result in failure ?
If not, let us revert this PR and simplify the code.

Reply:
Note that `createClient` failures of executor startup will not result in task fail. The executor will fail after max io retries, and then the app attempt will fail appMaster after the max number of executor failures is reached, and there are still many app attempts waiting for us to retry. It could be hours before you notice the app fail without the driver log telling what mistake we've made.

Reply:
@yaooqinn we have discussed this quite a bit already and now going in circles :-)
Essentially my point is this - instead of rejecting invalid configurations, we are trying to workaround invalid user configs. Once released, this will be a pattern we have to then continue supporting - I am not seeing a good reason why we have to introduce this new behavior.

+CC @dongjoon-hyun as well, since you reviewed the PR.